vllm/engine/arg_utils.py

```diff
@@ -368,12 +372,75 @@ class EngineArgs:
+        return StructuredOutputConfig.backend
+
+    @guided_decoding_backend.setter
+    def guided_decoding_backend(self, value: StructuredOutputBackend):
+        self.guided_decoding_backend = value
+
+    @property
+    @deprecated(
+        "`guided_decoding_disable_fallback` is deprecated and has been renamed to `structured_output_config.disable_fallback`. This will be removed in v0.10.0. Please use the `disable_fallback` argument in `structured_output_config` instead."
+    )
+    def guided_decoding_disable_fallback(self) -> bool:
+        return StructuredOutputConfig.disable_fallback
+
+    @guided_decoding_disable_fallback.setter
+    def guided_decoding_disable_fallback(self, value: bool):
+        self.guided_decoding_disable_fallback = value
+
+    @property
+    @deprecated(
+        "`guided_decoding_disable_any_whitespace` is deprecated and has been renamed to `structured_output_config.disable_any_whitespace`. This will be removed in v0.10.0. Please use the `disable_any_whitespace` argument in `structured_output_config` instead."
+    )
+    def guided_decoding_disable_any_whitespace(self) -> bool:
+        return StructuredOutputConfig.disable_any_whitespace
+
+    @guided_decoding_disable_any_whitespace.setter
+    def guided_decoding_disable_any_whitespace(self, value: bool):
+        self.guided_decoding_disable_any_whitespace = value
+
+    @property
+    @deprecated(
+        "`guided_decoding_disable_additional_properties` is deprecated and has been renamed to `structured_output_config.disable_additional_properties`. This will be removed in v0.10.0. Please use the `disable_additional_properties` argument in `structured_output_config` instead."
+    )
+    def guided_decoding_disable_additional_properties(self) -> bool:
+        return StructuredOutputConfig.disable_additional_properties
+
+    @guided_decoding_disable_additional_properties.setter
+    def guided_decoding_disable_additional_properties(self, value: bool):
+        self.guided_decoding_disable_additional_properties = value
+
+    @property
+    @deprecated(
+        "`enable_reasoning` is deprecated and has been renamed to `structured_output_config.enable_reasoning`. This will be removed in v0.10.0. Please check based on `reasoning_parser` instead."
+    )
+    def enable_reasoning(self) -> Optional[bool]:
+        return None
+
+    @enable_reasoning.setter
+    def enable_reasoning(self, value: Optional[bool]):
+        self.enable_reasoning = value
+
+    @property
+    @deprecated(
+        "`reasoning_parser` is deprecated and has been renamed to `structured_output_config.reasoning_backend`. This will be removed in v0.10.0. Please use the `reasoning_backend` argument in `structured_output_config` instead."
+    )
+    def reasoning_parser(self) -> str:
+        return StructuredOutputConfig.reasoning_backend
+
+    @reasoning_parser.setter
+    def reasoning_parser(self, value: str):
+        self.reasoning_parser = value
```

Comment:
All of these properties fetch the value from the class, not the instance.

They should be `self.structured_output_config.x` not `StructuredOutputConfig.x`